adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -321,7 +327,22 @@ private URL buildQueryString(final String instance, final String authorizationEn
         return new URL(builder.build().toString());
     }
 
+    /**
+     * @return {@link ReentrantLock} for locking the network request queue.
+     */
+    private ReentrantLock getLock() {
```

Comment:
This check feels a little weird for a few reasons. It behaves like a factory (in that it returns new instances) but `sLock` is not assigned here.

It also initializes a `static` field from an instance method and that feels strange.

I suspect that if the lock is not being acquired in one method and released in another, and there aren't any `waits` involved a more primitive synchronization technique may be sufficient (such as a synchronzied method or block, without the use of a `ReentrantLock`) -- I'll leave this up to your judgement but this looks dodgy

Reply:
synchronization method is tied to an object. If there are multiple discovery objects are created, it's not doing any synchronization actually.